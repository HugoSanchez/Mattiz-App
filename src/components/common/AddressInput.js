import React from 'react';
import { View, Dimensions } from 'react-native';
import { Input } from 'react-native-elements';

import CustomCard from './CustomCard';

import colors from '../../constants/colors';

// Dims. 
const width =  Dimensions.get('window').width

// Props:
// @ flex:          -Required. Container flex.
// @ address:       -Required. Input value.
// @ placeholder:   -Required. Placeholder.
// @ onChangeText:  -Required. Function to execute.

const AddressInput = props => {
    return (
        <View style={[ styles.container, { flex: props.flex } ]}>
            <CustomCard 
                elevated={true}
                style={ styles.cardStyle }>
                <Input 
                    placeholder={props.placeholder}
                    inputContainerStyle={ styles.inputContainer }
                    placeholderTextColor='#040026'
                    value={ props.address }
                    secureTextEntry={false}
                    onChangeText={ value => props.onChangeText(value) }
                    inputStyle={ props.address > 0 ? styles.textInput : styles.placeholder  }
                />
            </CustomCard>
        </View>
    );
}

const styles = {
    container: { 
        flex: 3, 
        alignSelf: 'stretch', 
        alignItems: 'center', 
        justifyContent: 'center' 
    },
    cardStyle: { 
        width: width * 0.7, 
        height: '100%', 
        alignSelf: 'center', 
        borderRadius: 10 
    },
    inputContainer: { 
        borderBottomColor: 'transparent', 
        marginTop: 5 
    },
    placeholder: {
        fontFamily: 'Aleo-Light', 
        fontSize: 18 
    },
    textInput: { 
        fontFamily: 'Aleo-Regular', 
        fontSize: 18, 
        color: colors.palleteDarkGreen 
    },
};

export default AddressInput;