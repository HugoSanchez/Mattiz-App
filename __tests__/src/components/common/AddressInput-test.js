import 'react-native'
import React from 'react'
import AddressInput from '../../../../src/components/common/AddressInput'
import {render, fireEvent} from '@testing-library/react-native'

test('Renders with title', () => {
	const {container} = render(<AddressInput />)
	expect(container.children[0]).toMatchSnapshot()
})
