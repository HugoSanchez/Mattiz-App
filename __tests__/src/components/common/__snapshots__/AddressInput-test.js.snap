// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with title 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 3,
        "justifyContent": "center",
      },
      Object {
        "flex": undefined,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#FDFFFE",
          "borderRadius": 15,
          "justifyContent": "center",
          "margin": 5,
        },
        Object {
          "alignSelf": "center",
          "borderRadius": 10,
          "height": "100%",
          "width": 525,
        },
        Object {
          "elevation": 7,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.29,
          "shadowRadius": 4.65,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "transparent",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "marginTop": 5,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          placeholderTextColor="#040026"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontFamily": "Aleo-Light",
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;
